### PR TITLE
GH-428 Add homes placeholders

### PR DESCRIPTION
**Pull request by failutee**
## Description
This change introduces new placeholders (#428)

## Placeholders
- **homes_owned** - Homes owned by the player
- **homes_count** - Number of homes owned by the player
- **homes_limit** - Limit of maximum homes for specific player

### CODE REVIEW

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/placeholder/HomesPlaceholderImpl.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    public HomesPlaceholderImpl(HomeManager homeManager) {
+        this.homeManager = homeManager;
+    }
+
+    @Override
+    public String apply(String text, Player targetPlayer) {
+        Collection<Home> homes = this.homeManager.getHomes(targetPlayer.getUniqueId());
+
+        if (homes.isEmpty()) {
+            return "You don't have any home.";
```
Comment (vLuckyyy) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/home/placeholder/HomesPlaceholderImpl.java:24]: It should be translatable

**Review by Jakubk15 — APPROVED** [review]

File: eternalcore-core/src/main/java/com/eternalcode/core/translation/Translation.java [hunk 2]
```diff
@@ -362,6 +362,13 @@ interface LanguageSection {
         Notice languageChanged();
     }
 
+    // Placeholder section
+    PlaceholderSection placeholder();
+
+    interface PlaceholderSection {
+        String homeNotFound();
+    }
```
Comment (vLuckyyy) [inline eternalcore-core/src/main/java/com/eternalcode/core/translation/Translation.java:370]: It seems to me that it should rather be in the home section. When someone uses, for example, only the home system, they configure only this section. In this case, he will have to scroll through one large configuration file

**Review by vLuckyyy — APPROVED** [review]

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/placeholder/HomeLimitGlobalPlaceholderImpl.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.eternalcode.core.feature.home.placeholder;
+
+import com.eternalcode.core.configuration.implementation.PluginConfiguration;
+import com.eternalcode.core.placeholder.PlaceholderReplacer;
+import org.bukkit.entity.Player;
+
+public class HomeLimitGlobalPlaceholderImpl implements PlaceholderReplacer {
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/home/placeholder/HomeLimitGlobalPlaceholderImpl.java:7]: The logic of this placeholder makes no sense to me, it returns the home limit for the highest rank in the config. I think it should be deleted

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/placeholder/HomeLimitPlaceholderImpl.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    @Override
+    public String apply(String text, Player targetPlayer) {
+        int allowedHomes = this.pluginConfiguration.homes.maxHomes.entrySet()
+            .stream()
+            .filter(entry -> targetPlayer.hasPermission(entry.getKey()))
+            .map(Map.Entry::getValue)
+            .max(Integer::compare)
+            .orElse(0);
+
+        return String.valueOf(allowedHomes);
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/home/placeholder/HomeLimitPlaceholderImpl.java:26]: This code is copy part of SetHomeCommand#getMaxAmountOfHomes method. Can you move this method to the HomeManager and then use there?

File: eternalcore-core/src/main/java/com/eternalcode/core/EternalCore.java [hunk 5]
```diff
@@ -310,8 +319,15 @@ public EternalCore(Plugin plugin) {
             .bukkitScheduler(plugin)
             .build();
 
-        this.placeholderRegistry.registerPlaceholder(PlaceholderReplacer.of("online", player -> String.valueOf(server.getOnlinePlayers().size())));
-        this.placeholderRegistry.registerPlaceholder(PlaceholderReplacer.of("afk", player -> String.valueOf(this.afkService.isAfk(player.getUniqueId()))));
+        Stream.of(
+            PlaceholderReplacer.of("online", player -> String.valueOf(server.getOnlinePlayers().size())),
+            PlaceholderReplacer.of("afk", player -> String.valueOf(this.afkService.isAfk(player.getUniqueId()))),
+
+            PlaceholderReplacer.of("homes", new HomesPlaceholderImpl(this.homeManager, this.userManager, this.translationManager)),
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/EternalCore.java:326]: homes -> homes_owned

File: eternalcore-core/src/main/java/com/eternalcode/core/EternalCore.java [hunk 6]
```diff
@@ -310,8 +319,15 @@ public EternalCore(Plugin plugin) {
...
+            PlaceholderReplacer.of("homes_number", new HomeNumberPlaceholderImpl(this.homeManager)),
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/EternalCore.java:327]: homes_number -> homes_count

**Review by Rollczi — CHANGES_REQUESTED** [review]

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/placeholder/HomeLimitGlobalPlaceholderImpl.java (hunk 3 above)
Comment (vLuckyyy) [reply eternalcore-core/src/main/java/com/eternalcode/core/feature/home/placeholder/HomeLimitGlobalPlaceholderImpl.java:7]: gif

File: eternalcore-core/src/main/java/com/eternalcode/core/feature/home/HomeManager.java [hunk 7]
```diff
@@ -67,6 +69,19 @@ public Collection<Home> getHomes(UUID user) {
         return Collections.unmodifiableCollection(this.homes.getOrDefault(user, new HashMap<>()).values());
     }
 
+    public int getMaxAmountOfHomes(Player player, Map<String, Integer> homes) {
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/feature/home/HomeManager.java:72]: Przekazuj sekcje od konfiguracji, mapki samej w sobie nie przekazujemy.

**Review by Rollczi — APPROVED** [review]

**Pull request merged**